nixos/modules/services/misc/nixos-needsreboot.nix

```diff
@@ -0,0 +1,24 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.nixos-needsreboot;
+in
+{
+  options = {
+    services.nixos-needsreboot = {
+      enable = lib.mkEnableOption "nixos-needsreboot, which determines if you need to reboot your NixOS machine";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.nixos-needsreboot = {
+      description = "Determine if you need to reboot your NixOS machine";
+      requires = [ "nixos-upgrade.service" ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        ExecStart = "${pkgs.nixos-needsreboot}/bin/nixos-needsreboot";
```

Comment:
We can use lib.getExe here

Reply:
I assumed that was for instances when the user had an option to select the package. Should I still use `lib.getExe` given that there's only one package?

Reply:
Yes.
```suggestion
ExecStart = lib.getExe pkgs.nixos-needsreboot;
```